lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -781,24 +781,48 @@ SILCombiner::visitInjectEnumAddrInst(InjectEnumAddrInst *IEAI) {
 
   // Ok, we have a payload enum, make sure that we have a store previous to
   // us...
-  SILBasicBlock::iterator II = IEAI->getIterator();
+  auto allocStackOperand = IEAI->getOperand();
   StoreInst *SI = nullptr;
   InitEnumDataAddrInst *DataAddrInst = nullptr;
   ApplyInst *AI = nullptr;
   Operand *EnumInitOperand = nullptr;
-  for (;;) {
-    if (II == IEAI->getParent()->begin())
+  for (auto usersIt = allocStackOperand->use_begin(); usersIt != allocStackOperand->use_end(); ++usersIt) {
+    auto currUser = usersIt->getUser();
+    // Find a Store whose destination is taken from an init_enum_data_addr
+    // whose address is same allocation as our inject_enum_addr.
+    DataAddrInst = dyn_cast<InitEnumDataAddrInst>(currUser);
```

Comment:
Yes. I spoke with @shajrawi offline about this. IIRC, we have run into this bug before (good we found it).